Framework: Bump recommended Node version to active LTS

aduth opened the pull request:
This pull request seeks to increase the recommended version of Node from 6.11.1 to the [active LTS release](https://github.com/nodejs/Release#release-schedule), currently 8.9.0 ([released 2017-10-31](https://nodejs.org/en/blog/release/v8.9.0/)).

__Testing instructions:__

Assuming you use [nvm](https://github.com/creationix/nvm) for managing Node versions, run `nvm install` and `nvm use` to switch to the LTS version. Otherwise install Node to v8.9.0. With a clean node modules (`rm -rf node_modules`) verify that `npm install`, `npm run build`, `npm run dev` complete with no issues.

Verify in Travis CI logs that Node is running 8.9.0 when defined as LTS:

```
3.53s$ nvm install && nvm use
Found '/home/travis/build/WordPress/gutenberg/.nvmrc' with version <lts/*>
Downloading and installing node v8.9.0...
Downloading https://nodejs.org/dist/v8.9.0/node-v8.9.0-linux-x64.tar.xz...
######################################################################## 100.0%
Computing checksum with sha256sum
Checksums matched!
Now using node v8.9.0 (npm v5.5.1)
Found '/home/travis/build/WordPress/gutenberg/.nvmrc' with version <lts/*>
Now using node v8.9.0 (npm v5.5.1)
```

https://travis-ci.org/WordPress/gutenberg/jobs/295819466

aduth committed:
Framework: Defer to nvmrc for Travis Node versions

aduth committed:
Framework: Bump recommended Node version to active LTS

gziolo commented on .travis.yml:
```diff
@@ -19,7 +19,7 @@ cache:
 before_install:
-  - nvm install 6.11.1 && nvm use 6.11.1
+  - nvm install && nvm use
   - npm install --global npm@5
```
Do we still need to override npm on Travis?

aduth replied:
It is probably safe to drop this, since Node 8.x will come with npm 5.x. It may be slightly different behavior in that this would install the latest 5.x, whereas I expect Node will come preinstalled with a pinned version of npm. I don't think we're specifically relying on this though.

ntwb replied:
npm's policy is that the latest release of npm is the supported release regardless of any _pinned version_ a Node.js release may ship.

gziolo reviewed (approved):
It works as advertised 👍 
Tested locally all commands listed + `npm test`.

aduth committed:
Framework: Drop explicit npm install

Previously needed since last LTS included npm 3.x. Since current LTS includes 5.x+, not necessary.

aduth commented:
Failing code coverage test on changes only to `.yml`, `.nvmrc`, and `.md` files 😅

gziolo commented:
Yes, I'm wondering if we can improve this integration as it errors too often without any reason.
